course7/homeworks/anton/ls/main.cc

```diff
@@ -0,0 +1,135 @@
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+
+namespace fs = boost::filesystem;
+namespace po = boost::program_options;
+
+struct configuration
+{
+    std::string file_path = ".";
+    bool show_help = false;
+    bool show_tree = false;
+    bool show_long = false;
+};
+
+configuration parse_args(int argc, char *argv[]);
+void print_file_tree(const fs::path& dir, const configuration& config, const std::string& filler = "");
+size_t get_file_size(const fs::path&);
+
+int main(int argc, char *argv[])
+{
+    try
+    {
+        const configuration& config = parse_args(argc, argv);
+
+        if (!config.show_help)
+        {
+            fs::path path(config.file_path);
+            print_file_tree(path, config);
+        }
+
+        return 0;
+    }
+    catch(std::exception& ex)
+    {
+        std::cout << "\033[1;31m" << ex.what() << "\033[0m" << std::endl;
```

Comment:
Don't.
If you let the exception pass trough, a correct core dump file can be created that contains the state of the process at the moment of _throwing_ the exception, including the location of the throw, stacktrace, memory etc.

Otherwise it's impossible to debug the problem

Reply:
aah, my understanding of core dumps was completely wrong, thank you for the explanation. But, anyway, I believe, that program should not be core dumped if the user provided a wrong file name, or there is an issue with access rights, etc. So i changed `std::exception` to `boost::filesystem::error`. Is it ok?